File: app/src/main/java/com/otraempresa/android/example/RequestTransferTokenAsyncTask.java
package com.otraempresa.android.example;

import android.content.Context;

import com.kushkipagos.android.Amount;
import com.kushkipagos.android.KushkiException;
import com.kushkipagos.android.Transaction;
import com.kushkipagos.android.Transfer;

public class RequestTransferTokenAsyncTask extends AbstractRequestTransferTokenAsyncTask {

    RequestTransferTokenAsyncTask(Context context) {
        super(context);
    }

    @Override
    protected Transaction requestTransferToken(Transfer token) throws KushkiException {
        return kushki.transferTokens(new Amount(12.2,0.0,1.2),"www.kushki.com",
                token.toJsonObject().optString("userType"),token.toJsonObject().optString("documentType"),
                token.toJsonObject().optString("documentNumber"),token.toJsonObject().optString("email"),
                "CLP");

    }

}

File: app/src/main/java/com/kushkipagos/kushkidemo/services/RequestTransferTokenAsyncTask.kt
package com.kushkipagos.kushkidemo.services

import android.content.Context
import com.kushkipagos.exceptions.KushkiException
import com.kushkipagos.kushkidemo.contracts.AbstractRequestTransferTokenAsyncTask
import com.kushkipagos.models.Amount
import com.kushkipagos.models.Transaction
import com.kushkipagos.models.Transfer

class RequestTransferTokenAsyncTask: AbstractRequestTransferTokenAsyncTask {
    constructor(context: Context) : super(context)

    @Throws(KushkiException::class)
    override fun requestTransferToken(token: Transfer): Transaction {
        return _kushki.requestTransferToken(
            Amount(12.2, 0.0, 1.2),
            "www.kushki.com",
            token.toJsonObject().optString("userType"),
            token.toJsonObject().optString("documentType"),
            token.toJsonObject().optString("documentNumber"),
            token.toJsonObject().optString("email"),
            "CLP"
        )
    }
}